build/circle-test.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -e
+
+outdir="${TMPDIR}"
+if [ "${CIRCLE_ARTIFACTS}" != "" ]; then
+    outdir="${CIRCLE_ARTIFACTS}"
+fi
+
+builder=$(dirname $0)/builder.sh
+
+# 1. Run "make check" to verify coding guidelines.
+echo "make check"
+time ${builder} make GITHOOKS= check | tee "${outdir}/check.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 2. Verify that "go generate" was run.
+echo "verifying generated files"
+time ${builder} /bin/bash -c "(go generate ./... && git ls-files --modified --deleted --others --exclude-standard | diff /dev/null -) || (git add -A && git diff -u HEAD && false)" | tee "${outdir}/generate.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 3. Run "make testrace".
+match='^panic|^[Gg]oroutine \d+|(read|write) by.*goroutine|DATA RACE'
+echo "make testrace"
+time ${builder} make GITHOOKS= testrace \
```

Comment:
That was one of the useful aspects of running through `circle.yml`, it would always run all of the test statements separately even if some failed. Maybe the `xml` conversion and posting an error can go into a `trap`.